assets/js/just-the-docs.js

```diff
@@ -31,7 +31,10 @@ function initNav() {
     }
     if (target) {
       e.preventDefault();
-      target.ariaPressed = target.parentNode.classList.toggle('active');
+      const active = target.parentNode.classList.toggle('active');
+      const passive = target.parentNode.classList.toggle('passive');
+      if (active && passive) target.parentNode.classList.toggle('passive');
```

Comment:
nit: wondering if the double-toggle is an issue with flashing/perf. What about

```suggestion
      if (active || target.parentNode.classList.contains('passive')) target.parentNode.classList.toggle('passive');
```

Reply:
I hadn't thought about the possibility of flashes. I have almost no experience of trying to implement DOM updates.

However, your suggestion appears to eliminate only one of the sources of double-toggling: wouldn't `toggle('active')` followed by `toggle{'passive')` also be a potential flash?

Reply:
`target.parentNode.classList` may contain `active` or `passive` or neither (never both).

As you may have discovered, the presence of the `passive` class in the DOM often has the same effect as the absence of the `active` class. The exception is for collections configured with `nav_fold: true`: when JS is disabled, they are neither `active` nor `passive`, but their items need to be displayed.; with JS enabled, selecting a page in one collection should fold all the others. I've not been able to get that effect without the use of `passive` (but I may have overlooked something).

The intended effect on `target.parentNode.classList` in `initNav` is:
1. blank changes to `active`
2. `active` changes to `passive`
3. `passive` changes to `active`
4. `active passive` never arises.

Using toggling, cases 2 and 3 both require double-toggling. Presumably it's possible to avoid that by atomically replacing the entire class-list, but I'm not familiar with the JS primitives for manipulating strings.

Moreover, `activateNav` currently involves multiple toggling in the nav panel. It would be good to know whether such toggling does actually cause flashes when navigating within/between pages before deciding whether to eliminate this particular double-toggling  or not.